### DISCUSSION TITLE
Root element

### DISCUSSION THREAD

**Discussion by MicTech** [ElMassimo/oj_serializers#16] [Q&A]
Hello 👋 
I'm experimenting with Oj::Serializer and our API using root elements in responses, even for a single record.

The issue I'm having is that I cannot make it work in Oj::Serializer. ☹️

Here is my serializer
```
class OjDocumentSerializer < Oj::Serializer
   attributes :id, :content
 end
```

Here is my controller
```
class Api::DocumentsController < Api::RestResourcesController
   def show
     @doc = current_tenant.documents.find_by(id: params[:id])

     render json: OjDocumentSerializer.one(@doc, root: :document)
   end
```

The response is without a root element
```
{"id" => "f7a8b63c-4eec-4505-ad7c-d0f10000315d", "content" => "Foo Bar"}
```

Thank you

**Comment by ElMassimo** [comment]
Hi Michal!

`root` should not be passed to the serializer, it should be a keyword passed to `render`, and it's only handled when using [`compat`](https://github.com/ElMassimo/oj_serializers/blob/main/MIGRATION_GUIDE.md#rendering-).

Otherwise, the recommended approach is to use a Hash:

```ruby
render json: {
  document: OjDocumentSerializer.one(@doc),
}
```

Not only is it more explicit and intuitive, it lends itself naturally to extension if you need to render additional properties.

**Reply by MicTech** [reply]
@ElMassimo Thank you. It makes more sense now 🙏 

I tried what you suggest and the problem with it is that I'm getting following response body

```
{"document":"{\"id\":\"f7a8b63c-4eec-4505-ad7c-d0f10000315d\",\"content\":\"Foo Bar\"}\n"}
```

Which isn't correct from my point of view 🤷‍♂️

**Reply by ElMassimo** [reply]
Seems odd, it looks like the serializer is encoded as a string. That's not the default behavior, as you can you can [see in this example app](https://github.com/ElMassimo/oj_serializers/blob/main/examples/my_api/app/controllers/albums_controller.rb#L5).

Are you using custom configuration for `oj`? Or a different gem that might override the default JSON renderer in Rails?

If you provide a minimal reproduction I can take a look.
